frontend/apps/web/components/onboarding-checklist/WelcomeDialog.tsx

```diff
@@ -0,0 +1,161 @@
+import {
+  getAccountOnboardingConfig,
+  setAccountOnboardingConfig,
+} from '@neosync/sdk/connectquery';
+import { DialogDescription } from '@radix-ui/react-dialog';
+import { useQueryClient } from '@tanstack/react-query';
+import { ReactElement, useEffect, useState } from 'react';
+import { toast } from 'sonner';
+import { useAccount } from '../providers/account-provider';
+import { Skeleton } from '../ui/skeleton';
+import Configure from './Configure';
+import Connect from './Connect';
+import StepProgress from './StepProgress';
+import Sync from './Sync';
+import WelcomeOverview from './WelcomeOverview';
+import WelcomeRouter from './WelcomeRouter';
+
+export interface FormStep {
+  name: string;
+  component: JSX.Element;
+}
+
+export default function WelcomeDialog(): ReactElement {
+  const { account } = useAccount();
+  const {
+    data,
+    isLoading,
+    isFetching: isValidating,
+    error,
+  } = useQuery(
+    getAccountOnboardingConfig,
+    { accountId: account?.id ?? '' },
+    { enabled: !!account?.id }
+  );
+  const queryclient = useQueryClient();
+  const { mutateAsync: setOnboardingConfigAsync } = useMutation(
+    setAccountOnboardingConfig
+  );
+
+  const [currentStep, setCurrentStep] = useState<number>(0);
+  const [isSubmitting, setIsSubmitting] = useState<boolean>(false);
+  const [showGuide, setShowGuide] = useState<boolean>(false);
+
+  const multiStepForm: FormStep[] = [
+    {
+      name: 'welcome',
+      component: (
+        <WelcomeOverview
+          currentStep={currentStep}
+          setCurrentStep={setCurrentStep}
+          setIsDialogOpen={setShowGuide}
+          completeForm={completeForm}
+        />
+      ),
+    },
+    {
+      name: 'connect',
+      component: (
+        <Connect currentStep={currentStep} setCurrentStep={setCurrentStep} />
+      ),
+    },
+    {
+      name: 'configure',
+      component: (
+        <Configure currentStep={currentStep} setCurrentStep={setCurrentStep} />
+      ),
+    },
+    {
+      name: 'execute',
+      component: (
+        <Sync currentStep={currentStep} setCurrentStep={setCurrentStep} />
+      ),
+    },
+    {
+      name: 'router',
+      component: (
+        <WelcomeRouter
+          currentStep={currentStep}
+          setCurrentStep={setCurrentStep}
+          setIsDialogOpen={setShowGuide}
+          completeForm={completeForm}
+        />
+      ),
+    },
+  ];
+
+  async function completeForm() {
+    if (!account?.id || isSubmitting) {
+      return;
+    }
+    setIsSubmitting(true);
+    try {
+      const resp = await setOnboardingConfigAsync({
+        accountId: account.id,
+        config: {
+          hasCompletedOnboarding: true,
+        },
+      });
+      queryclient.setQueryData(
+        createConnectQueryKey(getAccountOnboardingConfig, {
+          accountId: account.id,
+        }),
+        new GetAccountOnboardingConfigResponse({
+          config: resp.config,
+        })
+      );
+      setShowGuide(false);
```

Comment:
just wanted to put in here that this kinda sucks if for whatever reason they can't hit our server to set the onboarding config to true. it will stay stuck on the screen until our server comes back online. Not much we can do about taht I suppose unless we were to cache it in local storage or something. Maybe that is a task for another day however.

Reply:
yeah this could a follow up where we store it in local storage and check that if the db isn't active